llvm/include/llvm/Analysis/MemoryLocation.h

```diff
@@ -102,7 +102,7 @@ class LocationSize {
   //
   // Since the overwhelming majority of users of this provide precise values,
   // this assumes the provided value is precise.
-  constexpr LocationSize(uint64_t Raw)
+  explicit constexpr LocationSize(uint64_t Raw)
```

Comment:
Seeing as you've migrated the uses to LocationSize::precise(), maybe also resolve the FIXME by making this ctor private?